src/mongocxx/database.cpp

```diff
@@ -157,5 +164,12 @@ collection database::collection(stdx::string_view name) const {
     return mongocxx::collection(*this, name);
 }
 
+const std::unique_ptr<database::impl>& database::get_impl() const {
+    if(!_impl) {
+        throw operation_exception{make_error_code(error_code::k_database_uninitialized)};
```

Comment:
As operation_exception is the least common denominator, I think we should throw that here.  If other methods make calls that might trigger this exception, they can always catch and re-throw a new exception with a more specific type.